herald/herald/Sensor/BLE/BLEDatabase.swift

```diff
@@ -320,38 +320,59 @@ class BLEPseudoDeviceAddress {
         }}
     
     init?(fromAdvertisementData: [String: Any]) {
-        guard let manufacturerData = fromAdvertisementData["kCBAdvDataManufacturerData"] as? Data else {
+        guard let manufacturerData = fromAdvertisementData[CBAdvertisementDataManufacturerDataKey] as? Data else {
             return nil
         }
-        guard let manufacturerId = manufacturerData.uint16(0), manufacturerId == BLESensorConfiguration.manufacturerIdForSensor else {
+        guard let manufacturerId = manufacturerData.uint16(0) else {
             return nil
         }
-        guard manufacturerData.count == 8 else {
-            return nil
+        // HERALD pseudo device address
+        if manufacturerId == BLESensorConfiguration.manufacturerIdForSensor, manufacturerData.count == 8 {
```

Comment:
Please provide tests to ensure 6 byte pseudo mac address conversion to 8 byte Int64 always converts to/from without losing information and introducing clashes.

Reply:
Done. Comprehensive cross-platform encode/decode tests for pseudo device address.